Client/Frontend/Browser/BrowserPrompts.swift

```diff
@@ -112,6 +127,11 @@ struct TextInputAlert: JSAlertInfo {
         alertController.addAction(UIAlertAction(title: Strings.OKString, style: .default) { _ in
             self.completionHandler(input.text)
         })
+        if suppressHandler != nil {
```

Comment:
We really should find a way to abstract this, rather than having this logic pasted amongst three separate classes.  Even going as far as creating a new class `JSAlert: JSAlertInfo` that handles this, and having these three classes subclass that, rather than conforming to `JSAlertInfo` .

Otherwise a more simple option maybe to use a `UIAlertAction` on these classes instead of a `supressHandler`.

```
if let alertAction = alertAction {
  alertController.addAction(alertAction)
}
```

And the `runJavaScriptTextInputPanelWithPrompt` function would set an alert action with title and callback and everything, instead of each of these configuring it manually.